projects/Mallard/src/authentication/lib/Authorizer.ts

```diff
@@ -73,23 +74,47 @@ class Authorizer<
         this.checkUserHasAccess = checkUserHasAccess
     }
 
+    private logAuthCacheClear = (feature: Feature, reason?: string) => {
+        loggingService.log({
+            level: Level.INFO,
+            message: 'Clearing all authentication caches',
+            optionalFields: { reason, feature },
+        })
+    }
+
     private async handleAuthPromise(
         promise: Promise<AuthResult<T>>,
         connectivity: Connectivity,
     ) {
         let attempt: ResolvedAttempt<T>
+        const feature = Feature.SIGN_IN
         try {
             const result = await promise
 
             attempt = cataResult<T, ResolvedAttempt<T>>(result, {
                 valid: data => ValidAttempt(data, connectivity),
                 invalid: reason => {
+                    this.logAuthCacheClear(feature, reason)
                     this.clearCaches()
                     return InvalidAttempt(connectivity, reason)
                 },
-                error: reason => ErrorAttempt(connectivity, reason),
+                error: reason => {
+                    loggingService.log({
+                        level: Level.ERROR,
+                        message:
+                            'Authorization service has returned an error, error attempt created',
+                        optionalFields: { feature, reason },
+                    })
+                    return ErrorAttempt(connectivity, reason)
```

Comment:
Can we also add the `errorService.captureException` here? This will then log to Sentry as well.